prime-router/src/test/kotlin/validation/ValueValidationTest.kt

```diff
@@ -0,0 +1,69 @@
+package gov.cdc.prime.router.validation
+
+import assertk.assertThat
+import assertk.assertions.hasSize
+import assertk.assertions.isInstanceOf
+import io.konform.validation.Invalid
+import io.konform.validation.Valid
+import io.konform.validation.Validation
+import io.konform.validation.jsonschema.minItems
+import io.konform.validation.jsonschema.minLength
+import io.konform.validation.jsonschema.minimum
+import io.konform.validation.onEach
+import kotlin.test.Test
+
+class ValueValidationTest {
+
+    private val validation = Validation {
+        TestDataClass::string {
+            minLength(2)
+        }
+
+        TestDataClass::nullable required {
+            minimum(0)
+        }
+
+        TestDataClass::list {
+            minItems(2)
+            onEach {
+                minLength(2)
+            }
+        }
+    }
+
+    @Test
+    fun valid() {
+        val test = TestDataClass(
+            string = "string",
+            nullable = 5,
+            list = listOf(
+                "test",
+                "string"
+            )
+        )
+
+        val validated = validation.validate(test)
+        assertThat(validated).isInstanceOf<Valid<TestDataClass>>()
+    }
+
+    @Test
+    fun invalid() {
+        val test = TestDataClass(
+            string = "a", // to short
+            nullable = -5, // negative
+            list = listOf( // list not long enough
+                "1", // too short
+            )
+        )
+
+        val invalid = validation.validate(test)
+        assertThat(invalid).isInstanceOf<Invalid<TestDataClass>>()
+        assertThat(invalid.errors).hasSize(4)
```

Comment:
This is cool, it checks all of them instead of stopping after first error

Reply:
That was definitely a major perk of the library. Getting all the feedback you need after running the CLI tool once should be pretty helpful for the engagement team.